table/block_based/partitioned_filter_block.cc

```diff
@@ -79,7 +79,7 @@ void PartitionedFilterBlockBuilder::MaybeCutAFilterBlock(
   const bool add_prefix =
       next_key && prefix_extractor() && prefix_extractor()->InDomain(*next_key);
   if (add_prefix) {
-    FullFilterBlockBuilder::AddPrefix(*next_key);
+    AddKey(prefix_extractor()->Transform(*next_key));
```

Comment:
Why this change? It seems like we want to skip adding and `num_added_++` if redundant.

Reply:
Good finding.
We cannot call `AddPrefix` because it may update `lasts_prefix_str_` so that it stores the prefix of `*next_key`. However, when we build the next filter partition, we need the prefix of the last key of the previous filter partition.
Originally I thought adding the same string to the filter is OK, but maybe it's not. Maybe let me do the checking here before calling `AddKey()`.

Reply:
Oh right. It should be "safe" to add redundantly, though the filter implementation will allocate space for one more key than necessary.